### PR TITLE
Fix RtMidi leaks when using MidiIn or MidiOut (#256)

### PR DESCRIPTION
**Pull request by TheOtherDays**
Some remarks:

- https://github.com/mido/mido/blob/master/mido/backends/rtmidi.py#L119 and https://github.com/mido/mido/blob/master/mido/backends/rtmidi.py#L184
still use MidiIn and MidiOut

- examples might need to be fixed, I'm not sure

- other rtMidi-based backends might need some work too

### CODE REVIEW

**Review by olemb — APPROVED** [review]
Looks good.

**Comment by olemb** [comment]
Thanks!

* the `MidiIn` and `MidiOut` that are created here are cleaned up in `PortCommon._close()`. That's why I added that last `delete()`. (Sorry about the multiple inheritance. It makes the code a bit hard to follow.)
* as long as the code `port.close()` this will take care calling `delete()`. If the port is not closed the `MidiIn` or `MidiOut` object will hang around anyway.
* that's a good point. We should check if the `rtmidi_python` backend has the same problem. I'll add a new issue.

**Comment by TheNetStriker** [comment]
I've also just tested this fix and it works. I always had this problem because I watch for usb events and then attach the midi ports. Every time after 32 reconnects the following error occured:

ALSA lib seq_hw.c:466:(snd_seq_hw_open) open /dev/snd/seq failed: Cannot allocate memory
Traceback (most recent call last):
  File "src/_rtmidi.pyx", line 818, in rtmidi._rtmidi.MidiIn.__cinit__
RuntimeError: MidiInAlsa::initialize: error creating ALSA sequencer client object.

Here is my test script if anyone is interested:

```
import mido
import time

def searchMidiDevice():
    port_names = mido.get_input_names()
    for name in port_names:
        if "X-Touch One" in name:
            return name
    return None

midiPortname = searchMidiDevice()

inputMidi = None
outputMidi = None

try:
    i = 1
    while i < 100:
        if inputMidi != None:
            inputMidi.close()
            inputMidi = None
        if outputMidi != None:
            outputMidi.close()
            outputMidi = None

        print(i)
        i += 1
        inputMidi = mido.open_input(midiPortname)
        outputMidi = mido.open_output(midiPortname)
        time.sleep(2)
except OSError as e:
    print("connectMidi() Error: {}".format(e))
```

**Comment by cyberic99** [comment]
@olemb do you plan to release a new version anytime soon?

this leak/crash can really be a blocker for some uses

thanks!
